ConcernsCaseWork/ConcernsCaseWork.Tests/Pages/Team/SelectColleaguesTests.cs

```diff
@@ -0,0 +1,171 @@
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using ITeamsModelService = ConcernsCaseWork.Services.Teams.ITeamsModelService;
+
+namespace ConcernsCaseWork.Tests.Pages.Team
+{
+	[Parallelizable(ParallelScope.All)]
+	public class SelectColleaguesTests
+	{
+		[Test]
+		public async Task WhenOnGetAsync_Return_Page()
+		{
+			// arrange
+			var testFixture = new TestFixture()
+				.WithUsersAvailableForSelection("user1", "user2", "Mr.Bean")
+				.WithPreviouslySelectedUsers("Mr.Bean");
+
+			var sut = testFixture.BuildSut(authenticatedPage: true);
+
+			// act
+			var pageResponse = await sut.OnGetAsync();
+
+			// assert
+			Assert.That(pageResponse, Is.Not.Null);
+			Assert.That(sut.SelectedColleagues, Is.Not.Null);
+			Assert.That(sut.Users, Is.Not.Null);
+			Assert.That(sut.Users.Length, Is.EqualTo(3));
+			Assert.That(sut.TempData, Is.Empty);
+
+			testFixture.VerifyMethodEntered(nameof(SelectColleaguesPageModel.OnGetAsync));
+		}
+
+		[Test]
+		public async Task WhenOnPostSelectColleagues_UpdatesSelections_And_Redirects_To_TeamCaseworkPage()
+		{
+			// arrange
+			const string NewUsernameSelection = "Fred.Flintstone";
+			const string ExpectedRedirectUrl = "/#team-casework";
+
+			var testFixture = new TestFixture()
+				.WithPreviouslySelectedUsers("Mr.Bean");
+
+			var pageModel = testFixture.BuildSut(authenticatedPage: true);
+
+			// act
+			pageModel.SelectedColleagues = new List<string> { NewUsernameSelection };
+			var pageResponse = await pageModel.OnPostSelectColleagues();
+
+			// assert
+			var page = pageResponse as RedirectResult;
+			Assert.NotNull(page);
+			Assert.That(page, Is.Not.Null);
+			Assert.IsTrue(page.Url.Equals(ExpectedRedirectUrl));
+
+			testFixture.VerifyMethodEntered(nameof(SelectColleaguesPageModel.OnPostSelectColleagues));
+			testFixture.MockTeamsService.Verify(x => x.UpdateCaseworkTeam(It.Is<ConcernsTeamCaseworkModel>(m => m.OwnerId == testFixture.CurrentUserName
+				&& m.TeamMembers.Length == 1
+				&& m.TeamMembers[0] == NewUsernameSelection
+			)));
+		}
+
+		[Test]
+		public async Task WhenOnPostSelectColleagues_RequestForm_Missing_Return_ErrorOnPage()
+		{
+			// arrange			
+			var testFixture = new TestFixture()
+				.WithNoCurrentUser();
+			// act
+			var sut = testFixture.BuildSut(authenticatedPage: false);
+			var pageResponse = await sut.OnPostSelectColleagues();
+
+			// assert
+			var page = pageResponse as PageResult;
+			Assert.That(page, Is.Not.Null);
+			Assert.IsEmpty(sut.SelectedColleagues);
+			Assert.IsEmpty(sut.Users);
+			Assert.IsNotEmpty(sut.TempData);
+			Assert.That(sut.TempData["Error.Message"], Is.EqualTo("An error occurred posting the form, please try again. If the error persists contact the service administrator."));
```

Comment:
I'm assuming this isn't a user validation error but a more serious error? I'll check further in the code...

Reply:
To be honest this page was the old 2nd admin page repurposed, and this test was already there.
I think originally it was assumed it's possible to post to the page without a form, and so the name of the test mentioned a missing form.
I've renamed the test to be 'WithException' instead. I'm not sure not having a request is even something that we should worry about because we should be moving towards model binding